pkg/virt-handler/non-root.go

```diff
@@ -156,5 +179,8 @@ func (c *VirtualMachineController) nonRootSetup(origVMI *v1.VirtualMachineInstan
 	if err := c.prepareVFIO(res); err != nil {
 		return err
 	}
+	if err := c.prepareNetwork(origVMI, res); err != nil {
```

Comment:
This is called **after** the network setup.
This raises the question on why we need to own the `/net/tun`. The network setup is the one using it.
If it is about the "mask" thing, then we will have a problem here.

Reply:
AFAIU, the intention was to enable the virt-stack in the virt-launcher to access it.

> If it is about the "mask" thing, then we will have a problem here.

I don't follow, could you please elaborate?

Reply:
> AFAIU, the intention was to enable the virt-stack in the virt-launcher to access it.

I do not think libvirt needs to access the `tun` device directly, this is why virt-handler creates the `tap` "device" itself. But maybe I am wrong and it does somehow effect it.
I suggest to try it.



> > If it is about the "mask" thing, then we will have a problem here.

> I don't follow, could you please elaborate?

https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/configuring_basic_system_settings/managing-file-system-permissions_configuring-basic-system-settings#proc_displaying-the-current-value-of-the-umask_assembly_managing-the-umask

Reply:
I've posted https://github.com/kubevirt/kubevirt/pull/13959 as a draft to test this theory.

Reply:
@EdDev All network tests have passed without calming ownership over /dev/tun.
Can we leave it as a follow-up, or do you want to have the change on this PR?

Reply:
I do not understand if you considered nodes with umask set to be very restrictive.
IMO you need to check it locally with umask set to RO for the owner (or something like that).

@xpivarc was very worry about this in the past.

> Can we leave it as a follow-up, or do you want to have the change on this PR?

If you confirm that when the tun device has no access for the group/user and everything still works, you could do this in a follow up. As I see it, it needs to be checked first with umask set on the node.

Reply:
On my local kubevirtci setup, the umask is 0022.
After manually changing /dev/net/tun's permissions from 666 to 644 - connectivity between two VMs still works.

Reply:
Permissions were changed on the nodes.

Reply:
Also worked with 600.

Reply:
Can you try with 600 ?
Can you confirm the tun device is still 644 or 600 in the pod and the owner is root? (i.e. no one else changed it).

Reply:
`/dev/net/tun` in the virt-launcher's compute container is set to 666 and owned by root.
The umask in it is 0022.

Reply:
> `/dev/net/tun` in the virt-launcher's compute container is set to 666 and owned by root. The umask in it is 0022.

I think you cannot check this with

> After manually changing /dev/net/tun's permissions from 666 to 644 - connectivity between two VMs still works.

The umask needs to be changed and see it in the pod as well.

Reply:
OK, so the ownership is performed for virt-launcher/libvirt consumption. This should have no effect on virt-handler creating the tap device as root.

Therefore, this thread can be resolved.
This also means that we probably cannot remove the ownership step because libvirt seems to need it:
https://github.com/libvirt/libvirt/blob/b5dfcdc7ab9091e0a4467036a35e0dd9bbd64c8d/src/hypervisor/domain_interface.c#L110
https://github.com/libvirt/libvirt/blob/b5dfcdc7ab9091e0a4467036a35e0dd9bbd64c8d/src/util/virnetdevtap.c#L232
Anyway, it is not in this change scope.